TACHO_DRIVER.md

```diff
@@ -0,0 +1,199 @@
+Tacho Driver Update Proposal
+============================
+
+The tacho drivers work in most cases, but they have some defects against them
+
+Addresses the following issues in:
+
+- [ ] [ev3dev/ev3dev#488](https://github.com/ev3dev/ev3dev/issues/488)
+- [ ] [ev3dev/ev3dev#467](https://github.com/ev3dev/ev3dev/issues/467)
+- [ ] [ev3dev/ev3dev#455](https://github.com/ev3dev/ev3dev/issues/455)
+- [ ] [ev3dev/ev3dev#454](https://github.com/ev3dev/ev3dev/issues/454)
+- [ ] [ev3dev/ev3dev#453](https://github.com/ev3dev/ev3dev/issues/453)
+- [ ] [ev3dev/ev3dev#356](https://github.com/ev3dev/ev3dev/issues/356)
+- [ ] [ev3dev/ev3dev#224](https://github.com/ev3dev/ev3dev/issues/224)
+- [ ] [ev3dev/ev3dev#149](https://github.com/ev3dev/ev3dev/issues/149)
+- [ ] [ev3dev/ev3dev#148](https://github.com/ev3dev/ev3dev/issues/148)
+- [ ] [ev3dev/ev3dev#54](https://github.com/ev3dev/ev3dev/issues/54)
+
+Other tasks that need to be dealt with for this proposal:
+
+- [ ] Add LibreOffice Template for graphs - I'd like to make sure that we have
+a consistent format for the diagrams so that when they need changes we're not
+recreating them from scratch.
+- [ ] Add pointers to spec.json here so that we have a consistent notation
+for commands, functions, and mode. We may need to update `spec.json` so that
+the operation is better specified there.
+
+In order to properly fix the issues, and add the `sync` feature that users have
+been asking for, this document describes the operating modes and functions for
+the tacho motors.
+
+The `tacho-motor` drivers must strike a balance between doing the most common
+functions that a robot builder needs but not becoming so complex that the
+operation is unreliable or mysterious.
+
+## The sys Interface
+
+All motor operations are contolled by writing data to attributes within the
+`/sys/class/tacho-motor/motorN/` folder. There is one folder per motor.
+
+Some of the attributes are valid only during particular `run` modes - and with
+that in mind, the read and write operations can have the following return codes:
+
+| RETURN CODE  | DESCRIPTION
+| ------------ | -----------
+| 0            | Operation sucessful
+| ( -1) EPERM  | Operation not permitted (not the same as EACCESS!) 
+| (-13) EACCES | Permission denied
+| (-13) EBUSY  | Device or resource busy
+| (-22) EINVAL | Invalid argument
+
+The attributes operate as described in `spec.json` - with the following
+caveats for the `speed_regulation` value:
+
+| `speed_regulation` | Write `duty_cycle_sp` | Write `speed_sp`
+| ------------------ | --------------------- | ----------------
+| `off`              | Works                 | Return EPERM
+| `on`               | Return EPERM          | Works
```

Comment:
As long as we can tolerate people asking why changes to `duty_cycle_sp` are "not working" and it's because they are in `speed_regulation` in `on` mode :-)